lib/sippy_cup/scenario.rb

```diff
@@ -200,6 +200,62 @@ def invite(opts = {})
       @reference_variables += %w(remote_addr local_addr call_addr)
     end
 
+
+    #
+    # Send an invite message with auth header
+    #
+    # @param [Hash] opts A set of options to modify the message
+    # @option opts [Integer] :retrans
+    # @option opts [String] :headers Extra headers to place into the INVITE
+    def invite_auth(opts = {})
+      #opts[:retrans] ||= 500
+      # FIXME: The DTMF mapping (101) is hard-coded. It would be better if we could
+      # get this from the DTMF payload generator
+      from_addr = "#{@from_user}@#{@adv_ip}:[local_port]"
+      msg = <<-AUTH
+INVITE sip:#{to_addr} SIP/2.0 
+Via: SIP/2.0/[transport] #{@adv_ip}:[local_port];branch=[branch]
+From: "#{@from_user}" <sip:#{from_addr}>;tag=[call_number]
+To: <sip:#{to_addr}>
+Call-ID: [call_id]
+CSeq: [cseq] INVITE
+Contact: <sip:#{from_addr};transport=[transport]>
+[#{opts[:auth_keyword]}]
+Max-Forwards: 100
+User-Agent: #{USER_AGENT}
+Content-Type: application/sdp
+Content-Length: [len]
+#{opts.has_key?(:headers) ? opts.delete(:headers).sub(/\n*\Z/, "\n") : ''}
```

Comment:
you can use this ternary operator strategy used on this line to do something similar in the `invite` method and pass the `auth_keyword` into the options hash there and reduce duplication